### PR TITLE
feat: update role ux in projec access list

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/8512
Closes: https://github.com/lightdash/lightdash/issues/8499

### Description:

CleanShot 2024-01-22 at 14 50 19@2x

CleanShot 2024-01-22 at 14 50 05@2x

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Review by magnew — APPROVED** [review]
Reviewed and tested. I like how this is done. 

I left a non-blocking suggesting for code clarity. If you disagree feel free to leave as is.

File: packages/frontend/src/components/ProjectAccess/ProjectAccessRow.tsx [hunk 1]
```diff
@@ -68,88 +66,105 @@ const ProjectAccessRow: FC<Props> = ({
...
+                            onChange={(newRole: ProjectMemberRole) => {
+                                if (projectRole && projectRole.role) {
+                                    onUpdateAccess(newRole);
+                                } else {
+                                    onCreateAccess(newRole);
+                                }
+                            }}
+                        />
+
+                        {highestRole.type !== 'project' && projectRole?.role && (
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/components/ProjectAccess/ProjectAccessRow.tsx:93]: that makes sense. thank you!

**Comment by IrakliJani** [comment]
@TuringLovesDeathMetal @magnew addressed all the comments! thanks for the review!

CleanShot 2024-01-22 at 18 49 48

**Review by TuringLovesDeathMetal — APPROVED** [review]
😚 👌  amazing - this looks SO much better! Well done Irakli!

**Comment by IrakliJani** [comment]
ignoring e2e, is not related to the PR

**Pull request merged**